### DISCUSSION TITLE
Issue with currentAddress

### DISCUSSION THREAD

**Discussion by annetteshajan** [NationalSecurityAgency/ghidra#5955] [Q&A]
Hi
I am trying to develop a function on ghidra to track user activity. 
I understand from the docs that I can use _currentAddress_ to identify the address of the cursor location.
I am trying to use this to get the function name which the user has clicked on.

My code looks like this:
`while (!monitor.isCancelled()) {
         Listing listing = state.getCurrentProgram().getListing();
            Function func = listing.getFunctionContaining(currentAddress);
            if (func != null) {
                println("Selected Function: " + func.getName());
            }
            Thread.sleep(1000); `

However only the first address is getting stored here; it doesnt change dynamically while running.
Can you let me know how to get the value of currentAddress dynamically while the code is running?

**Comment by ryanmkurtz** [comment]
`currentAddress` (and its scripting friends) will only get updated when the `GhidraScript` is re-run.  It does not get updated in "real time".  Our scripts were not designed to be run in the background like you are doing.

You will most likely need to develop a `Plugin` as a Ghidra extension to get the results you are after.

**Reply by ryanmkurtz** [reply]
Notice how the bottom status bar updates the current address as you click around?  You would need to model your `Plugin` code after that.

**Comment by dragonmacher** [comment]
To say the same thing as @ryanmkurtz, with many more words:

Scripts were designed to run and complete, not to loop indefinitely.   The script is given the state at the time it is launched, which includes the current  address.   The script will do something useful for this current state and then exit.

Typically, people will invoke a script as needed when the script is intended to work on the current address.   You can add a keybinding to the script to make running it more convenient.   Basically, as you click around the Listing, you then run your script each time you want it to execute on the current location.

If you need something more advanced, then we suggest you write a plugin, which has a lifecycle that matches the tool.  Plugins can track the current address via the Ghidra event system, while scripts cannot easily do this.   Technically, you should be able to track events in your script using a listener on the `PluginTool`, but that is not something we recommend.
